framework/src/tinyalsa/tinyalsa.c

```diff
@@ -506,6 +504,18 @@ int pcm_writei(struct pcm *pcm, const void *data, unsigned int frame_count)
 				apb = (struct ap_buffer_s *)msg.u.pPtr;
 				memcpy(apb->samp, (char *)data + offset, nbytes);
 				apb->flags &= ~AUDIO_APB_OUTPUT_ENQUEUED;
+			} else if (msg.msgId == AUDIO_MSG_XRUN) {
```

Comment:
please use switch-case instead of if to improve readability.

Reply:
Sorry. But is it really needed to used switch for just 2 - 3 cases?
I feel it would make the code look complex since the switch will nest inside a while or for loop in some places. Lets keep current code for now?